app/javascript/controllers/area_served_controller.js

```diff
@@ -0,0 +1,29 @@
+import { Controller } from "@hotwired/stimulus"
+export default class extends Controller {
+    static targets= ["share", "total", "warning"]
+    connect () {
+        this.calculateClientShareTotal()
+    }
+    calculateClientShareTotal(){
+        let total = 0;
+        let share_targets = this.shareTargets
```

Comment:
This line isn't really necessary since the next line can just be `this.shareTargets.forEach(`